src/application/util/CryptoHelper.java

```diff
@@ -21,6 +21,8 @@
 import org.bouncycastle.crypto.params.KeyParameter;
 import org.bouncycastle.crypto.params.ParametersWithIV;
 
+import Exceptions.WrongPasswordException;
```

Comment:
revert all changes to this file.
the IllegalArgumentExceptions should be caught in the code calling decrypt which should then throw WrongPasswordException. why?
because this class is a helper class for multiple purposes and you couldnt possibly know if the password was the cause of a failed decrypt.